File: Adaline/C++/Neuron.cpp
#include "headers/Neuron.h"
#include <random>

Neuron::Neuron(int input_size) {
  std::random_device rd;
  std::mt19937 gen(rd());
  std::uniform_real_distribution<double> dist(-1.0, 1.0);

  this->b = dist(gen);
  for (int i = 0; i < input_size; i++) {
    this->w.push_back(dist(gen));
  }
}

Neuron::~Neuron() {}

double Neuron::soma(const std::vector<double> &x) {
  double u = 0.0;

  for (int i = 0; i < x.size(); ++i) {
    u += x[i] * this->w[i];
  }

  return u + this->b;
}

int Neuron::activation(const double u) { return (u >= 0) ? 1 : 0; }

File: Adaline/Python/src/neuron.py
import numpy as np 

class Neuron:
    def __init__(self, input_size:int):
        self.w = np.random.rand(input_size) * 2 - 1
        self.b = np.random.rand() * 2 - 1

    def soma(self, x):
        return np.dot(self.w, x) + self.b

    def activation(self, u): # Heaviside function
        if u >= 0:
            return 1
        else:
            return -1

    # Getters
    def getW(self) -> np.ndarray:
        return self.w
    def getB(self) -> float:
        return self.b

   # Setters
    def setW(self, w: np.ndarray):
        self.w = w
    def setB(self, b: float):
        self.b = b
